tests/imapp_benchmark/imapp_bench.go

```diff
@@ -0,0 +1,249 @@
+		exBytes = exBytes2
+
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+		}
+	}
+
+	result = exBytes
+}
+
+// // copied directly from github.com/ethereum/go-ethereum/core/vm/runtime/runtime.go
+// // so that we skip this in measured code
+// sets defaults on the config
+func setDefaults(cfg *runtime.Config) {
+	if cfg.ChainConfig == nil {
+		cfg.ChainConfig = &params.ChainConfig{
+			ChainID:             big.NewInt(1),
+			HomesteadBlock:      new(big.Int),
+			DAOForkBlock:        new(big.Int),
+			DAOForkSupport:      false,
+			EIP150Block:         new(big.Int),
+			EIP150Hash:          common.Hash{},
+			EIP155Block:         new(big.Int),
+			EIP158Block:         new(big.Int),
+			ByzantiumBlock:      new(big.Int),
+			ConstantinopleBlock: new(big.Int),
+			PetersburgBlock:     new(big.Int),
+			IstanbulBlock:       new(big.Int),
+			MuirGlacierBlock:    new(big.Int),
+			BerlinBlock:         new(big.Int),
+			LondonBlock:         new(big.Int),
+		}
+	}
+
+	if cfg.Difficulty == nil {
+		cfg.Difficulty = new(big.Int)
+	}
+	if cfg.Time == nil {
+		cfg.Time = big.NewInt(time.Now().Unix())
+	}
+	if cfg.GasLimit == 0 {
+		cfg.GasLimit = math.MaxUint64
+	}
+	if cfg.GasPrice == nil {
+		cfg.GasPrice = new(big.Int)
+	}
+	if cfg.Value == nil {
+		cfg.Value = new(big.Int)
+	}
+	if cfg.BlockNumber == nil {
+		cfg.BlockNumber = new(big.Int)
+	}
+	if cfg.GetHashFn == nil {
+		cfg.GetHashFn = func(n uint64) common.Hash {
+			return common.BytesToHash(crypto.Keccak256([]byte(new(big.Int).SetUint64(n).String())))
+		}
+	}
+	if cfg.BaseFee == nil {
+		cfg.BaseFee = big.NewInt(params.InitialBaseFee)
+	}
+}
+
+func runBenchmark(sampleSize int) {
+
+	bytecode = common.Hex2Bytes(bytecodeStore)
+
+	for i := 0; i < sampleSize; i++ {
+		r1 := testing.Benchmark(BenchmarkBytecodeExecutionSerial)
+		outputResults("BenchmarkBytecodeExecutionSerial", i, r1)
+
+		r2 := testing.Benchmark(BenchmarkBytecodeExecutionIndividualWithWarmup)
+		outputResults("BenchmarkBytecodeExecutionIndividualWithWarmup", i, r2)
+
+		r3 := testing.Benchmark(BenchmarkBytecodeExecutionIndividual)
+		outputResults("BenchmarkBytecodeExecutionIndividual", i, r3)
+	}
+}
+
+func outputResults(desc string, sampleId int, r testing.BenchmarkResult) {
+	fmt.Printf("%s,%d,%v,%v\n", desc, sampleId, r.N, r.NsPerOp())
+	fmt.Printf("%v, %v\n", r.AllocsPerOp(), r.AllocedBytesPerOp())
+}
+
+func runOverheadBenchmark(sampleSize int) {
+	for i := 1; i <= sampleSize; i++ {
+
+		bytecode = common.Hex2Bytes("00" + bytecodeStore)
+		rEmpty := testing.Benchmark(BenchmarkBytecodeExecutionSerial)
+
+		bytecode = common.Hex2Bytes(bytecodeStore)
+		rActual := testing.Benchmark(BenchmarkBytecodeExecutionSerial)
+
+		outputOverheadResults(i, rEmpty, rActual)
+	}
+}
+
+func outputOverheadResults(sampleId int, rEmpty testing.BenchmarkResult, rActual testing.BenchmarkResult) {
+	overheadTime := rEmpty.NsPerOp()
+	var executionLoopTime int64 = 0
+	var totalTime int64 = rEmpty.NsPerOp()
```

Comment:
I'm not a fan of this post processing. Can we simplify this to always be
```
overheadTime = rEmpty.NsPerOp()
executionLoopTime = rActual - rEmpty
totalTime = rActual
```
it's ok to have negative values, we'll know to watch out for these and we'll have a simpler job recovering the source measurements. Given that we're building the marginal models from it, I'll likely use `rActual` only, and use the `rEmpty` only for sanity checks.